src/palladio_fs/RulePackageFS.cpp

```diff
@@ -61,6 +61,14 @@ prtx::BinaryVectorPtr resolveRulePackageFile(const char* source, prt::Cache* cac
 	return {};
 }
 
+std::time_t getFileModificationTime(const char* path) {
```

Comment:
info.getModTime returns an int for HDK < 19.5 (see https://www.sidefx.com/docs/hdk/_h_d_k__changes_19_5.html)
I'd suggest checking for the houdini version and returning the corresponding type (like below in RulePackageInfoHelper::getModTime)
We could also consider defining a "hdk_time_t" or similar type based on the version in order to avoid a lot of #if directives duplication